### PR TITLE
Fix storage changelog entries for older releases for consistency (preview1, beta1) matching what shipped and add dates.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Hopefully, after this, we will get correct auto-version-increment PRs. See the following for more context:
https://github.com/Azure/azure-sdk-for-cpp/pull/1373#issuecomment-760018037

We also need to split up the storage release pipeline to be one per package and run them sequentially.

### CODE REVIEW

**Comment by ahsonkhan** [comment]
@msftbot make sure @katmsft or @Jinming-Hu gets to approve and require 2 sign offs.

**Comment by ahsonkhan** [comment]
@msftbot Only merge this pull request if it has 2 approvals.

**Comment by ahsonkhan** [comment]
@msftbot hold this pr for the next 12 hours

**Review by Jinming-Hu — APPROVED** [review]

**Review by vhvb1989 — APPROVED** [review]

**Pull request merged**
